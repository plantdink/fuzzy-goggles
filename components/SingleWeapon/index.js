export * from './SingleWeapon';
export { default } from './SingleWeapon';
