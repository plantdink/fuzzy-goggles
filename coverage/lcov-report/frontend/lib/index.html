
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> frontend/lib</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>20/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>39/46</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="abbreviationsDictionary.js"><a href="abbreviationsDictionary.js.html">abbreviationsDictionary.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="displayNumbers.js"><a href="displayNumbers.js.html">displayNumbers.js</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="displayStrings.js"><a href="displayStrings.js.html">displayStrings.js</a></td>
	<td data-value="84.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="74.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 74%"></div><div class="cover-empty" style="width: 26%"></div></div>
	</td>
	<td data-value="74.07" class="pct medium">74.07%</td>
	<td data-value="27" class="abs medium">20/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Apr 05 2022 07:32:49 GMT+1000 (Australian Eastern Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    