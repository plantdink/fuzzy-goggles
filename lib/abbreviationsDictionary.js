export function deAbbreviate(abbreviation) {
  let fullTerm = '';
  if (abbreviation === 'OOC') fullTerm = 'Damage to Targets out of Cover';
  if (abbreviation === 'DTH') fullTerm = 'Damage to Health';
  if (abbreviation === 'DTA') fullTerm = 'Damage to Armour';
  if (abbreviation === 'CHC') fullTerm = 'Critical Hit Chance';
  if (abbreviation === 'CHD') fullTerm = 'Critical Hit Damage';
  if (abbreviation === 'HSD') fullTerm = 'Head Shot Damage';
  return fullTerm;
}
