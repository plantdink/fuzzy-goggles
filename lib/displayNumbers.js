export function displayPercentage(number = 0) {
  const displayNumber = number / 10;
  return displayNumber;
}

export function hsDisplay(number = 0) {
  const displayNumber = number / 100;
  return displayNumber;
}

export function millisecondsToSeconds(number = 0) {
  const displayNumber = ((number % 60000) / 1000).toFixed(2);
  return displayNumber;
}

export function humanReadableNumber(num = 0, lang = null) {
  if (!num) return;
  const locale = lang || 'en'; // removed || document.documentElement.lang
  const number = parseFloat(num, 10);
  return number.toLocaleString(locale);
}
