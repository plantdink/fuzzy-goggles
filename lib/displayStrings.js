export function lowerCaseFirstLetter(string) {
  const lowerCaseString = string.toLowerCase();
  const titleString = lowerCaseString.replace(
    /(^\w{1})|(\s+\w{1})/g,
    (letter) => letter.toUpperCase()
  );
  const noWhiteSpaceString = titleString.replaceAll(/\s/g, '');
  const finalLink =
    noWhiteSpaceString.charAt(0).toLowerCase() + noWhiteSpaceString.slice(1);
  return finalLink;
}

export function titleCase(string) {
  return string
    .toLowerCase()
    .split(' ')
    .map(function (word) {
      return word.charAt(0).toUpperCase() + word.slice(1);
    })
    .join(' ');
}

export function stringToParagraphs(string) {
  const sentences = string.split(/(?<=\.)/g);
  return sentences.map((sentence, index) => <p key={index}>{sentence}</p>);
}

export function lowerCaseLink(string) {
  const stringToEdit = string[0].toLowerCase() + string.slice(1) || string;
  return stringToEdit;
}
