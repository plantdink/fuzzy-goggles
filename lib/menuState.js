import { createContext, useContext, useState } from 'react';

const LocalStateContext = createContext();
const LocalStateProvider = LocalStateContext.Provider;

function MenuStateProvider({ children }) {
  const [menuOpen, setMenuOpen] = useState(false);

  function toggleMenu() {
    setMenuOpen(!menuOpen);
  }

  function closeMenu() {
    setMenuOpen(false);
  }

  function openMenu() {
    setMenuOpen(true);
  }

  return (
    <LocalStateProvider
      value={{
        menuOpen,
        setMenuOpen,
        toggleMenu,
        closeMenu,
        openMenu,
      }}
    >
      {children}
    </LocalStateProvider>
  );
}

// custom hook for accessing menu local state
function useMenu() {
  const all = useContext(LocalStateContext);
  return all;
}
export { MenuStateProvider, useMenu };
