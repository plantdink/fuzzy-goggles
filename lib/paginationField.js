import { WEAPON_PAGINATION_QUERY } from '../queries/PaginationQueries';

export default function paginationField() {
  // apollo will ask read for items.
  // It will return them from the cache if they exist or
  // it will return false & trigger a network request
  return {
    keyArgs: false, // toggle for apollo to let this function handle things
    read(existing = [], { args, cache }) {
      const { skip, first } = args;

      // read the number of items on the page from the cache
      const data = cache.readQuery({ query: WEAPON_PAGINATION_QUERY });
      const count = data?._allWeaponsMeta?.count;
      const page = skip / first + 1;
      const pages = Math.ceil(count / first);

      // check for existing items
      const items = existing.slice(skip, skip + first).filter((x) => x);

      // if there are items & there are not enough for the requested amount & it is the last page, go to the network
      if (items.length && items.length !== first && page === pages) {
        return items;
      }
      // there are no items, request from network
      if (items.length !== first) {
        return false;
      }

      // if there are items in cache, return them
      if (items.length) {
        return items;
      }

      // fallback to network
      return false;
    },

    // apollo will run this when the network request returns with items
    merge(existing, incoming, { args }) {
      const { skip, first } = args;
      const merged = existing ? existing.slice(0) : [];
      for (let i = skip; i < skip + incoming.length; ++i) {
        merged[i] = incoming[i - skip];
      }

      // return the merged items from the cache
      return merged;
    },
  };
}
