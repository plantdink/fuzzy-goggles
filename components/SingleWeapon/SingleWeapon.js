import React from 'react';
import { gql } from '@apollo/client';
import styled from 'styled-components';
import {
  displayPercentage,
  hsDisplay,
  millisecondsToSeconds,
  humanReadableNumber,
} from '../../lib/displayNumbers';
import { deAbbreviate } from '../../lib/abbreviationsDictionary';

export const SingleWeaponStyles = styled.div`
  position: relative;
  display: flex;
  flex-flow: column wrap;
  max-width: var(--maxWidth);
  background: white;
  border: 1px solid var(--offWhite);
  box-shadow: var(--bs);
  padding: 2rem;
  margin-bottom: 3rem;
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;

  .single-weapon__content {
    display: flex;
    max-width: var(--maxWidth);
    justify-content: center;
    align-items: top;
    gap: 2rem;
    background: white;
    img {
      width: 100%;
      object-fit: contain;
    }
  }

  .single-weapon__sub-content {
    display: flex;
    justify-content: space-between;
  }

  .single-weapon__details {
    flex-direction: column;
    margin-bottom: 2rem;
  }

  .single-weapon__title-bar {
    transform: skew(7deg);
    margin-bottom: 1rem;
  }

  .single-weapon__heading {
    margin: 0 1rem;
    margin-top: -3rem;
    background: var(--orange);
    display: inline;
    line-height: 1.3;
    font-size: clamp(2rem, 5vw, 4rem);
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  .single-weapon__section-heading {
    margin: 0 1rem;
    margin-top: -3rem;
    background: var(--orange);
    display: inline;
    line-height: 1.3;
    font-size: clamp(2rem, 4vw, 3.5rem);
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  .single-weapon__subheading {
    margin: 0 1rem;
    /* margin-top: 0rem; */
    background: var(--orange);
    display: inline;
    line-height: 2;
    font-size: clamp(1rem, 3vw, 2rem);
    font-weight: 600;
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  ul {
    margin: 0;
    color: var(--text);
    font-weight: 400;
    font-size: 1.6rem;
    padding: 0;
    list-style-type: none;
  }

  li {
    margin: 0.75rem 0;
    /* padding: 0.5rem 0 0 1.5rem; */
    padding: 0 2.4rem;
  }

  p {
    color: var(--text);
    margin: 0.75rem 0;
    font-weight: 400;
    padding: 0 2.4rem;
    font-size: 1.6rem;
    line-height: 1.3;
  }

  table {
    table-layout: fixed;
    width: 100%;
    border-collapse: collapse;
    color: var(--text);
    line-height: 1.3;
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
  }

  th {
    font-weight: 500;
    /* width: 100%; */
  }

  th,
  td {
    padding: 0.5rem 2rem;
    text-align: left;
    vertical-align: baseline;
  }

  /* thead th:nth-child(1) {
    width: 20%;
  }

  thead th:nth-child(2) {
    width: 20%;
  }

  thead th:nth-child(3) {
    width: 20%;
  }

  thead th:nth-child(4) {
    width: 20%;
  } */

  blockquote {
    color: var(--text);
    margin: 0.85rem 0;
    padding: 0 2.4rem;
    font-weight: 300;
    font-size: 1.6rem;
  }

  small {
    color: var(--text);
    font-size: 1.2rem;
    padding: 0 2.4rem;
  }

  .named-weapon {
    background: var(--namedGold);
  }

  .exotic-weapon {
    background: var(--exoticOrange);
  }

  @media (max-width: 35em) {
    padding: 0;
    .single-weapon__content {
      padding-top: 1rem;
      padding-bottom: 1rem;
      display: flex;
      flex-direction: column;
    }

    .single-weapon__sub-content {
      flex-direction: column;
    }
  }
`;

export const SINGLE_WEAPON_QUERY = gql`
  query SINGLE_WEAPON_QUERY($id: ID!) {
    allWeapons(where: { id: $id }) {
      id
      class
      model
      family
      magazineSize
      rpm
      modSlots
      reloadSpeed
      reloadSpeedFromEmpty
      accuracy
      stability
      optimalRange
      maxRange
      headshotMultiplier
      weaponBonusType
      maxBonusValue
      weaponBonusTypeTwo
      maxBonusTwoValue
      damageLevel40
      damageWt5
      flavourText
      isNamed
      isExotic
      weaponTalent {
        id
        name
        image {
          id
          image {
            publicUrlTransformed
          }
        }
      }
      exoticWeaponAttachment {
        id
        name
        attachmentSlot
        opticsSlotBonusType
        opticsBonusValueLevel40
        opticsValueWT5
        magazineSlotBonusType
        magazineValueLevel40
        magazineValueWT5
        underbarrelSlotBonusType
        underbarrelValueLevel40
        underbarrelValueWT5
        muzzleSlotBonusType
        muzzleValueLevel40
        muzzleValueWT5
      }
      averageWeapon {
        id
        class
        classBonusType
        maxClassBonusValue
        modSlots
        rpm
        magazineSize
        headshotMultiplier
        accuracy
        stability
        optimalRange
        maxRange
        reloadSpeed
        reloadSpeedFromEmpty
        damageLevel40
        damageWT5
        weaponTalent {
          id
          name
          image {
            id
            image {
              publicUrlTransformed
            }
          }
        }
        weaponThirdAttribute {
          id
          attribute
          valueLevel40
          valueWT5
          classARAttribute
          classLMGAttribute
          classMMRAttribute
          classPistolAttribute
          classRifleAttribute
          classShotgunAttribute
          classSMGAttribute
        }
      }
    }
  }
`;

export default function SingleWeapon({ weapon }) {
  return (
    <>
      <div className="single-weapon__content">
        <div className="single-weapon__details">
          <div />
          <table>
            <thead className="single-weapon__title-bar">
              <tr>
                <th className="single-weapon__subheading">Base Damage</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope="row">Level 40</th>
                <td>{humanReadableNumber(weapon.damageLevel40)}</td>
              </tr>
              <tr>
                <th scope="row">World Tier 5</th>
                <td>{humanReadableNumber(weapon.damageWt5) || 'N/A'}</td>
              </tr>
            </tbody>
            <thead className="single-weapon__title-bar">
              <tr>
                <th className="single-weapon__subheading">Specifications</th>
              </tr>
            </thead>

            <tbody>
              {(!weapon.accuracy || !weapon.stability) && (
                <>
                  <tfoot>
                    <tr>
                      <td colSpan="2">
                        <small>
                          0 indicates no community-sourced value available
                        </small>
                      </td>
                    </tr>
                  </tfoot>
                </>
              )}

              {weapon.maxBonusValue > 1 && (
                <>
                  <tr>
                    <th scope="row">{deAbbreviate(weapon.weaponBonusType)}</th>
                    <td>{weapon.maxBonusValue} %</td>
                  </tr>
                </>
              )}

              {weapon.maxBonusTwoValue > 1 && (
                <>
                  <tr>
                    <th scope="row" data-testid="secondDamageBonusTest">
                      {deAbbreviate(weapon.weaponBonusTypeTwo)}
                    </th>
                    <td>{displayPercentage(weapon.maxBonusTwoValue)} %</td>
                  </tr>
                </>
              )}

              <tr>
                <th scope="row">Headshot Multiplier</th>
                <td>x {hsDisplay(weapon.headshotMultiplier)}</td>
              </tr>
              <tr>
                <th scope="row">Rate of Fire</th>
                <td>{weapon.rpm} rounds per minute</td>
              </tr>
              <tr>
                <th scope="row">Magazine</th>
                <td>{weapon.magazineSize}</td>
              </tr>

              {weapon.modSlots === 0 && (
                <>
                  <tr>
                    <th scope="row">Mod Slots</th>
                    <td>Nil adjustable</td>
                  </tr>
                </>
              )}

              {weapon.modSlots > 0 && (
                <>
                  <tr>
                    <th scope="row">Mod Slots</th>
                    <td>{weapon.modSlots}</td>
                  </tr>
                </>
              )}

              <tr>
                <th scope="row">Accuracy</th>
                <td>{displayPercentage(weapon.accuracy)} %</td>
              </tr>
              <tr>
                <th scope="row">Stability</th>
                <td>{displayPercentage(weapon.stability)} %</td>
              </tr>
              <tr>
                <th scope="row">Optimal Range</th>
                <td>{weapon.optimalRange} m</td>
              </tr>
              <tr>
                <th scope="row">Max Range</th>
                <td>{weapon.maxRange} m</td>
              </tr>
              <tr>
                <th scope="row">Reload Speed</th>
                <td>{millisecondsToSeconds(weapon.reloadSpeed)} seconds</td>
              </tr>
              <tr>
                <th scope="row">Reload Speed (empty magazine)</th>
                <td>
                  {millisecondsToSeconds(weapon.reloadSpeedFromEmpty)} seconds
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </>
  );
}
