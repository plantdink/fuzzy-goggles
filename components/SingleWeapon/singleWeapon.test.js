import { render, screen } from '@testing-library/react';
import SingleWeapon from '.';
import { fakeWeapon, fakeAverageWeapon } from '../../lib/testUtils';
import {
  humanReadableNumber,
  hsDisplay,
  displayPercentage,
  millisecondsToSeconds,
} from '../../lib/displayNumbers';
import { deAbbreviate } from '../../lib/abbreviationsDictionary';

const weapon = fakeWeapon();
const avgWeapon = fakeAverageWeapon();

describe('<SingleWeapon />', () => {
  it('renders the data correctly for a single weapon', async () => {
    const { container, debug } = render(
      <SingleWeapon weapon={weapon} avgWeapon={avgWeapon} />
    );
    // debug();

    expect(container).toMatchSnapshot();

    const weaponDamageLvl40 = humanReadableNumber(weapon.damageLevel40);
    const testLvl40WeaponDamage = screen.getByText(weaponDamageLvl40);
    expect(testLvl40WeaponDamage).toBeInTheDocument();

    const weaponDamageWT5 = humanReadableNumber(weapon.damageWt5);
    const testWT5WeaponDamage = screen.getByText(weaponDamageWT5);
    expect(testWT5WeaponDamage).toBeInTheDocument();

    //  Class Bonus Damage type & value
    const weaponTestBonusDamage = screen.getByRole('rowheader', {
      name: /damage to armour/i,
    });
    expect(weaponTestBonusDamage).toHaveTextContent(
      deAbbreviate(weapon.weaponBonusType)
    );
    const weaponTestBonusDamageValue = screen.getByRole('cell', {
      name: /21 %/i,
    });
    expect(weaponTestBonusDamageValue).toHaveTextContent(
      `${weapon.maxBonusValue} %`
    );

    // weapon HSD multiplier title & value
    const weaponTestHSDMultiplier = screen.getByRole('rowheader', {
      name: /headshot multiplier/i,
    });
    expect(weaponTestHSDMultiplier).toHaveTextContent('Headshot Multiplier');
    const weaponTestHSDMultiplierValue = screen.getByRole('cell', {
      name: /x 1\.55/i,
    });
    expect(weaponTestHSDMultiplierValue).toHaveTextContent(
      hsDisplay(weapon.headshotMultiplier)
    );

    //  weapon rpm heading & value
    const weaponTestRPM = screen.getByRole('row', {
      name: /rate of fire/i,
    });
    expect(weaponTestRPM).toHaveTextContent(`${weapon.rpm} rounds per minute`);

    //  weapon mag heading & size
    const weaponTestMagazineSize = screen.getByRole('row', {
      name: /magazine 30/i,
    });
    expect(weaponTestMagazineSize).toHaveTextContent(weapon.magazineSize);

    // weapon mod slots
    const weaponTestModSlots = screen.getByRole('row', {
      name: /mod slots/i,
    });
    expect(weaponTestModSlots).toHaveTextContent(weapon.modSlots);

    // weapon accuracy
    const weaponTestAccuracy = screen.getByRole('row', { name: /accuracy/i });
    expect(weaponTestAccuracy).toHaveTextContent(
      displayPercentage(weapon.accuracy)
    );

    // weapon stability
    const weaponTestStability = screen.getByRole('row', { name: /stability/i });
    expect(weaponTestStability).toHaveTextContent(
      displayPercentage(weapon.stability)
    );

    // weapon optimal range
    const weaponTestOptimalRange = screen.getByRole('row', {
      name: /optimal range/i,
    });
    expect(weaponTestOptimalRange).toHaveTextContent(
      `${weapon.optimalRange} m`
    );

    const weaponTestMaxRange = screen.getByRole('row', { name: /max range/i });
    expect(weaponTestMaxRange).toHaveTextContent(`${weapon.maxRange} m`);

    // weapon reload speed
    const weaponTestReloadSpeed = screen.getByRole('cell', {
      name: /2\.00 seconds/i,
    });
    expect(weaponTestReloadSpeed).toHaveTextContent(
      `${millisecondsToSeconds(weapon.reloadSpeed)} seconds`
    );

    // weapon empty mag reload speed
    const weaponTestEmptyReloadSpeed = screen.getByRole('cell', {
      name: /2\.50 seconds/i,
    });
    expect(weaponTestEmptyReloadSpeed).toHaveTextContent(
      `${millisecondsToSeconds(weapon.reloadSpeedFromEmpty)} seconds`
    );

    if (weapon.accuracy || weapon.stability > 0) {
      const small = container.querySelector('small');
      expect(small).not.toBeInTheDocument();
    }

    // it('Renders a Radar Chart', () => {
    //   const wrapper = mount(
    //     <RadarChart />
    //   )
    //   expect(wrapper).toMatchSnapshot();
  });
});
