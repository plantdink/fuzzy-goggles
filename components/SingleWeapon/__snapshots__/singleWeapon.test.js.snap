// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleWeapon /> renders the data correctly for a single weapon 1`] = `
<div>
  <div
    class="single-weapon__content"
  >
    <div
      class="single-weapon__details"
    >
      <div />
      <table>
        <thead
          class="single-weapon__title-bar"
        >
          <tr>
            <th
              class="single-weapon__subheading"
            >
              Base Damage
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              Level 40
            </th>
            <td>
              53,124
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              World Tier 5
            </th>
            <td>
              11,770
            </td>
          </tr>
        </tbody>
        <thead
          class="single-weapon__title-bar"
        >
          <tr>
            <th
              class="single-weapon__subheading"
            >
              Specifications
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              Damage to Armour
            </th>
            <td>
              21
               %
            </td>
          </tr>
          <tr>
            <th
              data-testid="secondDamageBonusTest"
              scope="row"
            >
              Damage to Health
            </th>
            <td>
              9.5
               %
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Headshot Multiplier
            </th>
            <td>
              x 
              1.55
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Rate of Fire
            </th>
            <td>
              650
               rounds per minute
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Magazine
            </th>
            <td>
              30
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Mod Slots
            </th>
            <td>
              4
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Accuracy
            </th>
            <td>
              61.8
               %
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Stability
            </th>
            <td>
              76
               %
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Optimal Range
            </th>
            <td>
              27
               m
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Max Range
            </th>
            <td>
              105
               m
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Reload Speed
            </th>
            <td>
              2.00
               seconds
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              Reload Speed (empty magazine)
            </th>
            <td>
              2.50
               seconds
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
